go/api/base_client.go

```diff
@@ -282,6 +300,41 @@ func (client *baseClient) executeCommandWithRoute(
 	return payload.value, nil
 }
 
+func (client *baseClient) getStatistics() (*C.struct_CommandResponse, error) {
```

Comment:
Why create a private function that is just wrapped by the public `GetStatistics()` function? This only complicates the code and forces devs to make a code hop to try and understand the full method. This private method is not reused anywhere else and has a 1:1 relationship with the GetStatistics method. I believe merging all of this logic directly into the public method would simplify code and make it easier to read and follow.